### ISSUE TITLE
FITS submitted from vrfy.sh

### ISSUE THREAD

**Issue by RussTreadon-NOAA** [NOAA-EMC/global-workflow#1058]
**Expected behavior**
`config.vrfy` allows the user to specify the `QUEUE` to use for the FITS job.   
```
if [ $VRFYFITS = "YES" ]; then

    export fit_ver="newm.1.3"
    export fitdir="$BASE_GIT/verif/global/Fit2Obs/${fit_ver}/batrun"
    export PRVT=$HOMEgfs/fix/gsi/prepobs_errtable.global
    export HYBLEVS=$HOMEgfs/fix/am/global_hyblev.l${LEVS}.txt
    export CUE2RUN=$QUEUE
```
Therefore, expect the FITS job to run in the user specified queue.

**Current behavior**
While `subfit_orion_netcdf` sets `CUE2RUN` based on user input
```
ACCOUNT=${ACCOUNT:-CFS-T2O}
CUE2RUN=${CUE2RUN:-dev}
KEEPDATA=${KEEPDATA:-NO}
```
it does not use the user set `CUE2RUN`.  Instead it hardwires the FITS job to use the batch queue.
```
#!/bin/bash --login
#SBATCH --job-name=FITS.$EXPNAM.$CDATE --time=20:00
#SBATCH --nodes=1 --ntasks=4
#SBATCH --exclusive
#SBATCH --output=$COMDAY/FITS.$EXPNAM.$CDATE.$$
#SBATCH --account=$ACCOUNT
#SBATCH --partition=orion
#SBATCH --qos batch
```

**Machines affected**
Orion

**To Reproduce**
Set up the `develop` g-w on Orion.   As a test, set `QUEUE=debug` in `config.base`.   Submit `gdasvrfy`.   The FITS job will be submitted to the batch queue even though `CUE2RUN=QUEUE=debug` is exported to `subfits_orion_netcdf` .  

**Additional Information**
Script `subits_orion_netcdf` resides [NOAA-EMC/Fit2Obs](https://github.com/NOAA-EMC/Fit2Obs).  Resolution of this issue will require changes in the Fit2Obs repo.    The issue is opened here since g-w submits the FITS job.

**Comment by RussTreadon-NOAA** [comment]
Tagging @jack-woollen for awareness.

**Comment by jack-woollen** [comment]
@RussTreadon-NOAA created tag newm.1.6 which accepts and honors CUE2RUN as either batch or debug in subfits_orion.

**Comment by RussTreadon-NOAA** [comment]
Thank you @jack-woollen .   I do not see tag newm.1.6 in [https://github.com/NOAA-EMC/Fit2Obs/tags](https://github.com/NOAA-EMC/Fit2Obs/tags).   I only see tags up to new.1.5.    Where do I find tag newm.1.6?

I see that [newmaster](https://github.com/NOAA-EMC/Fit2Obs/tree/newmaster) was updated at about the same time as your comment above.   For the time being, I installed newmaster in my space on Orion and updated my lo-res parallel to use this installation. 

In doing so, I discovered that while g-w job gdasvrfy has been submitting fit2obs, the job on Orion, the job has not been running.   This remains true with newmaster.   Here's what I see in the fit2obs job log file

```
+ slurm_script[12]: set +x
/var/spool/slurmd/job7198160/slurm_script: line 9: PS1: unbound variable

CommandNotFoundError: Your shell has not been properly configured to use 'conda deactivate'.
To initialize your shell, run

    $ conda init <SHELL_NAME>

Currently supported shells are:
  - bash
  - fish
  - tcsh
  - xonsh
  - zsh
  - powershell

See 'conda init --help' for more information and options.

IMPORTANT: You may need to close and restart your shell after running 'conda init'.
```

This is confusing.  I do not have conda in my `.bashrc`.

I'm going to stop here.   

@KateFriedman-NOAA , have we confirmed that the existing newm.1.3 fit2obs successfully runs to completion on Orion from the current head of g-w develop?

**Comment by jack-woollen** [comment]
@RussTreadon-NOAA Sorry I hadn't pushed the tag, but you picked up the changes which had been pushed to the repo. You should be able to see the tag now. I don't understand the error either. I don't find PS1 in any of the fit2obs scripts.

**Comment by WalterKolczynski-NOAA** [comment]
`$PS1` is the shell variable that controls what the prompt is. When conda is activated, it tries to modify that prompt. I'm not sure exactly why conda is being activated here, but the submitted job doesn't seem to have that. Maybe the module or environment are being exported to the submitted job somehow. I don't have the PS1 message in my own logs, but I also have a custom PS1 set in my .bashrc. Do you have the PS1 error in any other logs (I don't know that it would be fatal). It's also possible it is being masked in other jobs, since we pipe the output of the module load to `/dev/null`.

More generally, I don't know why a job is submitting another job on its own independently of the workflow manager anyway. This means the job isn't controlled by the workflow and is effectively orphaned as soon as it is submitted.

**Comment by RussTreadon-NOAA** [comment]
Thank you, @jack-woollen , for pushing the tag to the repo.  I installed newm.1.6 on Orion in `/work/noaa/da/Russ.Treadon/git/verif/global/Fit2Obs/newm.1.6`.   

Using newm.1.6 with the current head of g-w develop requires a few changes to `config.vrfy`
```
Orion-login-3[7] rtreadon$ pwd
/work/noaa/da/Russ.Treadon/para_gfs/prgsida4
Orion-login-3[8] rtreadon$ diff /work/noaa/da/Russ.Treadon/git/global_workflow/develop/parm/config/config.vrfy ./config.vrfy
31,32c31,33
<     export fit_ver="newm.1.3"
<     export fitdir="$BASE_GIT/verif/global/Fit2Obs/${fit_ver}/batrun"
---
>     export fit_ver="newm.1.6"
>     export BASE_GIT_DEV="/work/noaa/da/Russ.Treadon/git"
>     export fitdir="$BASE_GIT_DEV/verif/global/Fit2Obs/${fit_ver}/batrun"
45c46
<         export PREPQFITSH="$fitdir/subfits_hera_slurm"
---
>         export PREPQFITSH="$fitdir/subfits_hera"
47c48
<         export PREPQFITSH="$fitdir/subfits_s4_slurm"
---
>         export PREPQFITSH="$fitdir/subfits_s4"
49c50
<         export PREPQFITSH="$fitdir/subfits_orion_netcdf"
---
>         export PREPQFITSH="$fitdir/subfits_orion"
```

The `BASE_GIT_DEV` change is only required because I can not add newm.1.6 to  `BASE_GIT=/work/noaa/global/glopara/git`   (Tagging @KateFriedman-NOAA for awareness.)

I reran gdasvrfy for 2021122712.   This submits fit2obs for 2021122612.   The fit2obs job failed as before
```
+ slurm_script[12]: set +x
/var/spool/slurmd/job7199574/slurm_script: line 9: PS1: unbound variable

CommandNotFoundError: Your shell has not been properly configured to use 'conda deactivate'.
To initialize your shell, run

    $ conda init <SHELL_NAME>

Currently supported shells are:
  - bash
  - fish
  - tcsh
  - xonsh
  - zsh
  - powershell

See 'conda init --help' for more information and options.

IMPORTANT: You may need to close and restart your shell after running 'conda init'.
```
I see that g-w `modulefiles/module_base.orion.lua` loads `hpc-miniconda3/4.6.14`.   This module is in the job environment when `subfits_orion` submits the fit2obs job.    I think the default for sbatch is to export all environment variables.   As a test I added `--export=NONE` to the sbatch line in `subfits_orion`.   
```
/work/noaa/da/Russ.Treadon/git/verif/global/Fit2Obs/newm.1.6/batrun
Orion-login-3[66] rtreadon$ git diff subfits_orion
diff --git a/batrun/subfits_orion b/batrun/subfits_orion
index 897da72..dd739d4 100755
--- a/batrun/subfits_orion
+++ b/batrun/subfits_orion
@@ -32,7 +32,7 @@ set -euax
 vrfytmpdiris=$TMPDIR
 unset TMPDIR

-cat<<EOF | sbatch
+cat<<EOF | sbatch --export=NONE
 #!/bin/bash --login
 #SBATCH --job-name=FITS.$EXPNAM.$CDATE --time=20:00
 #SBATCH --nodes=1 --ntasks=3
@@ -60,6 +60,7 @@ export EXP=$EXPNAM
 export COMPONENT=${COMPONENT:-atmos}
 export COM_IN=$ROTDIR
 export KEEPDATA=$KEEPDATA
+export RUN_ENVIR=$RUN_ENVIR

 export fitdir=$fitdir
 export ARCDIR=$ARCDIR
```
This gets me past the conda error.   Note, I added the `RUN_ENVIR` line because this was not defined when fit2obs ran.   Even with the above changes, the fit2obs job failed at
```
+ eval COMLIC=
+ COMLIC=''
+ eval COMPRP=
+ COMPRP=''
+ COMPRP=''
+ set -u
+ [[ netcdf == nemsio ]]
+ [[ netcdf == netcdf ]]
+ [[ netcdf == nemsio ]]
+ suffix=nc
+ PRPI=/gdas.t12z.prepbufr
+ export PRPI
+ PRPO=/work/noaa/stmp/rtreadon/RUNDIRS/prgsida4/2021122712/gdas/prgsida4.2021122612.fit2obstmp/fitx/gdas.t12z.prepqa
+ export PRPO
+ PRPF=/work/noaa/stmp/rtreadon/RUNDIRS/prgsida4/2021122712/gdas/prgsida4.2021122612.fit2obstmp/fitx/gdas.t12z.prepqf
+ export PRPF
+ sig1=/gdas.t12z.atmanl.nc
+ export sig1
+ sfc1=/gdas.t12z.atmanl.nc
+ export sfc1
+ CNVS=/gdas.t12z.cnvstat
+ export CNVS
+ [ YES '!=' NO ]
+ /work/noaa/da/Russ.Treadon/git/verif/global/Fit2Obs/newm.1.6/batrun/../ush/ACprof
+ PRPI='/work/noaa/da/Russ.Treadon/git/verif/global/Fit2Obs/newm.1.6/batrun/../ush/ACprof missing /gdas.t12z.prepbufr or /gdas.t12z.prepbufr.acft_profiles'
```
Note that `PRPI`, `PRPO`, `PRPF`, `sig1`, `sfc1`, and `CNVS` are incomplete paths.   This, in part, is due to the fact that `COM_INA` and `COM_PRP` are not defined.   `subfits_orion` only defines `COM_IN`

We likely need a new issue to fully integrate newm.1.6 or a follow-on tag into g-w.

**Comment by jack-woollen** [comment]
@RussTreadon-NOAA As I recall I worked with Lin to run the fits with his parallel some time ago resulting in the latest version of subfits_orion. Since COM_INA and COM_PRP and COM_INF can vary we defined them in the calling script. Seems like that's the better place to define these locations.

**Comment by RussTreadon-NOAA** [comment]
@jack-woollen , I see what you say.   `subfits_orion` and `subfits_hera` differ from `subfits_wcoss2` in more than just SBATCH -vs- PBS.   I'm going to stop here with fit2obs for the time being.   I need to move onto GDASApp tasks.   Hopefully others can pick up this thread and iron out wrinkles on orion (and, presumably hera).

**Comment by lgannoaa** [comment]
May I know the status of this issue? Do we still need help?

**Comment by jack-woollen** [comment]
@lgannoaa what kind of help do you need?

**Comment by KateFriedman-NOAA** [comment]
FYI, right before I took leave for Thanksgiving I began testing a fixed version of fit2obs newm.1.5 on WCOSS2 (mainly needed conversion to bash due to recent script changes in global-workflow `develop`). I plan to send these changes back to the fit2obs repo shortly. Need to test on R&Ds as well.

This work doesn't address how we currently launch the fit2obs job via `vrfy.sh` though. Need to assign a developer (myself or other) to pull the fit2obs out of the vrfy job and into it's own job submitted via rocoto. Shouldn't be difficult to do.

**Comment by jack-woollen** [comment]
Hi @KateFriedman-NOAA . Just a heads up, the latest tag is newm.1.6.

**Comment by KateFriedman-NOAA** [comment]
> the latest tag is newm.1.6

Good to know, thanks @jack-woollen ! I'll restart my testing with a copy of that tag.

I was thinking I will make a branch of fit2obs `develop` and send my changes back via a PR. Let me know if you'd like the changes differently.

**Comment by jack-woollen** [comment]
Yes, please do.

**Comment by DavidHuber-NOAA** [comment]
@KateFriedman-NOAA @RussTreadon-NOAA Was this fixed by #1456?

**Comment by KateFriedman-NOAA** [comment]
Yes, good catch, PR #1456 should have also closed this. I have added mention of this issue to the top comment in that PR (for documentation) and am now closing this as complete via #1456. Thanks @DavidHuber-NOAA !
